src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOSupport.scala

```diff
@@ -39,4 +42,20 @@ trait ElasticSearchDAOSupport extends LazyLogging {
     }
   }
 
-}
+  def makeMapping(attributeJson: String): String = {
+    val definition = attributeJson.parseJson.convertTo[AttributeDefinition]
+    val maps = definition.properties flatMap { case (label: String, detail: AttributeDetail) =>
+      detailFromAttribute(label, detail)
+    }
```

Comment:
could you use `map` instead of `flatMap` here, and then have the `detailFromAttribute` cases return `(label -> value)` instead of `Map(label -> value)`?

In other words, you're currently creating a map of maps, then flattening it out; instead, you could just transform the entries.